evm_loader/program/src/allocator/solana/state_account_allocator.rs

```diff
@@ -0,0 +1,46 @@
+use std::alloc::Layout;
+use std::ptr::NonNull;
+use std::slice;
+
+use linked_list_allocator::Heap;
+
+use crate::allocator::solana::Alloc;
+use crate::allocator::STATE_ACCOUNT_DATA_ADDRESS;
+
+#[derive(Clone, Copy)]
+pub struct AccountAllocator;
+
+#[allow(clippy::cast_possible_truncation)] // HEAP_START_ADDRESS < usize::max
+                                           // Configure State/Holder Account heap: the offset of the heap object is at HEAP_OBJECT_PTR address.
+const HEAP_OBJECT_PTR: usize = STATE_ACCOUNT_DATA_ADDRESS + crate::account::HEAP_OFFSET_PTR;
+
+impl Alloc for AccountAllocator {
+    fn heap() -> &'static mut Heap {
+        let heap_object_offset_ptr = HEAP_OBJECT_PTR as *const usize;
+        let heap_object_offset = unsafe { std::ptr::read_unaligned(heap_object_offset_ptr) };
```

Comment:
`HEAP_OBJECT_PTR` currently aligned.
It has a constant address and we can guarantee with the compile time check that it will be aligned in the future.
If pointer immediately after the Holder's header is not aligned, you can just move it to the right.

`fn heap()` is called at every memory allocation and reading the unaligned pointer is not performance friendly

Reply:
Ok, keeping it as is for now.
We will return to this when we get the profiler start spending more time on evm optimization.